REACT_STYLEGUIDE.MD

```diff
@@ -45,14 +44,31 @@ Bad:
     };
   },
   onClick() {
-    this.setState({ clickCount: this.state.clickCount++ });
+    this.setState(prevState => ({
+      clickCount: prevState.clickCount + 1
+    }));
   },
-  render({ onClick }) {
+  render() {
     return <button className="kuiButton" onClick={this.onClick}/>
   }
 });
 ```
 
+### When a state change only involves the previous state, pass setState a function instead of an object.
```

Comment:
@weltenwort @chrisronline are you saying you think the rule should be always use a function?  sounds fine to me if so, just trying to figure out if you want me to change something here.